### DISCUSSION TITLE
Why doesn't lockFileMaintenance's addLabels work?

### DISCUSSION THREAD

**Discussion by rushilsrivastava** [renovatebot/renovate#10802] [Request Help] [answered]
I'm currently giving Renovate a shot and messing around with the `lockFileMaintenance` configuration settings to add specific labels for when a `lockFileMaintenance` PR is created. Here is what I have for my config.

<details>
  <summary>Renovate Config</summary>
  
```JSON
{
  "extends": [
    ":separateMajorReleases",
    ":ignoreUnstable",
    ":combinePatchMinorReleases",
    ":prImmediately",
    "group:monorepos",
    "group:recommended",
    ":ignoreModulesAndTests",
    ":updateNotScheduled"
  ],
  "timezone": "America/Los_Angeles",
  "schedule": [
    "after 5pm every weekday",
    "every weekend",
    "before 9am every weekday"
  ],
  "dependencyDashboard": true,
  "labels": ["type: dependencies"],
  "stabilityDays": 3,
  "lockFileMaintenance": {
    "enabled": true,
    "automerge": true,
    "addLabels": ["type: maintenance"]
  }
}
```
  
</details>

For whatever reason, lockFileMaintenance PRs only adds the `type: dependencies` label. I can't see any relevant logs on the Web App either. Can I only add labels via `packageRules`?

**Comment by viceice** [comment, marked as answer]
yes, please try a packageRule like:

```json
{
    "matchUpdateTypes": ["lockFileMaintenance"],
    "enabled": true,
    "automerge": true,
    "addLabels": ["type: maintenance"]
}
```

this should add `type: dependencies` and `type: maintenance` to the pr. if you want to fully override labels, you need to use `labels` openion instead of `addLabels`.

https://docs.renovatebot.com/configuration-options/#labels
https://docs.renovatebot.com/configuration-options/#addlabels

**Reply by rushilsrivastava** [reply]
Awesome - thanks so much!

**Comment by rarkins** [comment]
Labels are only applied to PRs at creation time. They won't be applied to any existing PRs.
